### ISSUE TITLE
Adding group support

### ISSUE THREAD

**Issue by JPchico** [aiidaplugins/aiida-lammps#32]
One nice feature of LAMMPS is the capability of treating different atoms in a different way via the definitions of groups. 

Currently this is not possible in aiida-lammps. Adding groups would be a very good addition to bring more flexibility to the kind of calculations that can be done. 

Part of this is addressed by the flexibilization of the input #27 . However, one would need to be able to pass which atoms belong to which group, one way would be via the kind_names in the `StructureData` where one could encode if an atom belongs to a given group or to another. One could also pass this as a separate `Dict`, or `ArrayData`.

The capability of having groups also means that fixes and computes need to be passed by group, meaning that this is something that also has to be dealt with in #27.

**Comment by JPchico** [comment]
Looking at what is available in the lammps documentation about the [read_data](https://docs.lammps.org/read_data.html) it does not seem possible to specify the group type directly in the file that is being read, one can only place all the atoms in that file in a given group.

It would seem that the "simplest" option is to allow the user to pass which atom types/ids belong to a given group as explained [here](https://docs.lammps.org/group.html). This could be done in the structure block.

The user could pass a dictionary such as 
```
structure_options = {
    'groups':{
        'name': 'foo', 'args':  ['id', 10, 25],
        'name': 'bar', 'args': ['type', 4],
        'name': 'foobar', 'args': ['id', '100:10000:10']
    }
}
```
The names of the groups would be stored in the input generator so that when setting fixes and/or computes the existence of the group is checked against.

Using the types and/or ids would give quite a bit of flexibility to the users, and avoid the issue of having to deal with changes to the way in which the lammps structure file is written and/or coming up with some complex derived structure type.

**Comment by JPchico** [comment]
Closed via #50
